mm2src/mm2_main/src/lp_ordermatch/orderbook_rpc.rs

```diff
@@ -137,9 +151,23 @@ pub async fn orderbook_rpc(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, S
     if base_ticker == rel_ticker && base_coin_conf["protocol"] == rel_coin_conf["protocol"] {
         return ERR!("Base and rel coins have the same orderbook tickers and protocols.");
     }
+    let base_coin_protocol: CoinProtocol = try_s!(json::from_value(base_coin_conf["protocol"].clone()));
```

Comment:
This doesn't seem required. You can make `is_mine_zhtlc` a new version of `is_my_order` and call it unconditionally as it was.